osquery/logger/logger.cpp

```diff
@@ -479,6 +479,18 @@ size_t queuedStatuses() {
   return BufferedLogSink::get().dump().size();
 }
 
+void waitLogRelay() {
+  if (kOptBufferedLogSinkSender.has_value()) {
+    /* NOTE: We are not doing a workaround for Windows
+       as in BufferedLogSink::WaitTillSent because we are not and we must not be
+       in a path called by Google Log, and failing to properly wait
+       for the thread to finish will either cause a race condition or a deadlock
+     */
+    kOptBufferedLogSinkSender->wait();
```

Comment:
Is is possible for this to block?

Reply:
The intent here **is** to block, to ensure the ordering of the threads; if you mean to ask "can this block **indefinitely**", if there's a bug, yes.

Reply:
Ha, yes _indefinitely_ was implied. In your estimate, how likely is a bug there? I imagine a kill-9 will still kill off osquery, and someone will report it and we can deal.

Reply:
> 

Well, that's difficult to estimate, but that shouldn't stop us, meaning, the synchronization is needed.